server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -235,10 +244,7 @@ private void mergeIndexResponses(
         List<FieldCapabilitiesFailure> failures = indexFailures.build(indexResponses.keySet());
         if (indexResponses.size() > 0) {
             if (request.isMergeResults()) {
-                // fork off to the management pool for merging the responses as the operation can run for longer than is acceptable
-                // on a transport thread in case of large numbers of indices and/or fields
-                threadPool.executor(ThreadPool.Names.SEARCH_COORDINATION)
-                    .submit(ActionRunnable.supply(listener, () -> merge(indexResponses, task, request, new ArrayList<>(failures))));
+                ActionListener.completeWith(listener, () -> merge(indexResponses, task, request, new ArrayList<>(failures)));
```

Comment:
NB before this change we would have rejected the `merge` call if the queue was full, but now we are calling this from a response handler which is force-executed, bypassing the queue-length restriction. IMO that's ok, but thought it worth calling out.